cds/modules/records/tasks.py

```diff
@@ -403,18 +403,14 @@ def _format_report(report):
         return '\n'.join(lines)
 
     cache = current_cache.get('task_missing_subformats:details') or {}
-    if 'last_run' not in cache:
-        # Save the last week date to have a default value the first time
-        cache['last_run'] = datetime.utcnow() - timedelta(days=7)
+    if 'end_date' not in cache:
+        # Set the end date to 7 days ago
+        cache['end_date'] = datetime.utcnow() - timedelta(days=7)
 
-    query = _filter_by_last_created(
+    record_uuids = _filter_by_last_created(
                 _get_all_records_with_bucket(),
-                start_date or cache['last_run'],
-                end_date)
-
-    record_uuids = query \
```

Comment:
You have removed this part, is that ok?

Reply:
Yes, because we already do `.group_by(RecordMetadata.id).with_entities(RecordMetadata.id)` on line `158`.